tests/std/tests/P0896R4_ranges_algorithm_machinery/test.cpp

```diff
@@ -983,3 +984,128 @@ namespace gh_1089 {
+    STATIC_ASSERT(sentinel_for<sentinel_archetype<sentinel_status::good>, iterator_archetype<iterator_status::input>>);
+
+    // Validate _No_throw_sentinel_for
+    STATIC_ASSERT(!_No_throw_sentinel_for<sentinel_archetype<sentinel_status::not_sentinel>,
+                  iterator_archetype<iterator_status::input>>);
+    STATIC_ASSERT(
+        _No_throw_sentinel_for<sentinel_archetype<sentinel_status::good>, iterator_archetype<iterator_status::input>>);
+    STATIC_ASSERT(!_No_throw_sentinel_for<iterator_archetype<iterator_status::input>,
+                  iterator_archetype<iterator_status::input>>);
+    STATIC_ASSERT(_No_throw_sentinel_for<iterator_archetype<iterator_status::forward>,
+        iterator_archetype<iterator_status::forward>>);
+
+    // Validate _No_throw_forward_iterator
+    STATIC_ASSERT(!_No_throw_forward_iterator<iterator_archetype<iterator_status::not_input>>);
+    STATIC_ASSERT(!_No_throw_forward_iterator<iterator_archetype<iterator_status::not_lvalue_reference>>);
+    STATIC_ASSERT(!_No_throw_forward_iterator<iterator_archetype<iterator_status::different_reference_and_value>>);
+    STATIC_ASSERT(!_No_throw_forward_iterator<iterator_archetype<iterator_status::input>>);
+    STATIC_ASSERT(_No_throw_forward_iterator<iterator_archetype<iterator_status::forward>>);
+
+    enum class range_status : int { not_range, not_input, input, forward };
+
+    template <range_status I>
+    struct range_archetype {
+        using It = std::conditional_t<I == range_status::not_range, void,
+            std::conditional_t<I == range_status::not_input, iterator_archetype<iterator_status::not_input>,
+                std::conditional_t<I == range_status::forward, iterator_archetype<iterator_status::forward>,
+                    iterator_archetype<iterator_status::input>>>>;
+        using Se = std::conditional_t<I == range_status::not_range, void,
+            std::conditional_t<I == range_status::forward, iterator_archetype<iterator_status::forward>,
+                std::default_sentinel_t>>;
+
+        It begin() const;
+        Se end() const;
+    };
+    // Verify range_archetype
+    STATIC_ASSERT(!ranges::range<range_archetype<range_status::not_range>>);
+    STATIC_ASSERT(ranges::range<range_archetype<range_status::not_input>>);
+    STATIC_ASSERT(ranges::range<range_archetype<range_status::input>>);
+    STATIC_ASSERT(ranges::range<range_archetype<range_status::forward>>);
+
+    // Validate _No_throw_input_range; note that the distinction betweeen range<R> and
+    // no-throw-sentinel-for<sentinel_t<R>, iterator_t<R>> is purely semantic, so we can't test them separately.
+    STATIC_ASSERT(!_No_throw_input_range<range_archetype<range_status::not_range>>);
+    STATIC_ASSERT(!_No_throw_input_range<range_archetype<range_status::not_input>>);
+    STATIC_ASSERT(_No_throw_input_range<range_archetype<range_status::input>>);
+    STATIC_ASSERT(_No_throw_input_range<range_archetype<range_status::forward>>);
+

```

Comment:
```suggestion

    // Validate _No_throw_forward_range
```
for consistency with the rest of the `STATIC_ASSERT` blocks.

Reply:
Weird, I must have ran out of `/`s by the time I got here. Fixed.